config/routes.rb

```diff
@@ -23,7 +23,10 @@
   end
   resources :watermarks
   resources :contacts
-  resources :contact_details, only: [:update]
+
+  get 'contact_details', to: 'contact_details#show'
+  put 'contact_details', to: 'contact_details#update'
+
```

Comment:
User resource instead of manual routes like `resources :contact_details, only: [:show, :update]`